org-code-javabuilder/lib/src/main/java/org/code/javabuilder/JavaBuilder.java

```diff
@@ -0,0 +1,25 @@
+package org.code.javabuilder;
+
+import java.io.PrintStream;
+
+/** The orchestrator for code compilation and execution. */
+public class JavaBuilder {
+  private final OutputAdapter outputAdapter;
+  private final InputAdapter inputAdapter;
+
+  public JavaBuilder(InputAdapter inputAdapter, OutputAdapter outputAdapter) {
+    this.outputAdapter = outputAdapter;
+    this.inputAdapter = inputAdapter;
+  }
+
+  /**
+   * Sets replaces System.in and System.out with our custom implementation and executes the user's
+   * code.
+   */
+  public void runUserCode() {
+    System.setOut(new PrintStream(new OutputRedirectionStream(this.outputAdapter), true));
```

Comment:
Maybe consider having the OutputRedirectionStream extend PrintStream (so we don't need the call to PrintStream here)?  In particular, the `true` parameter here is very important to this working the way you expect so it might be good to encapsulate that in the OutputRedirectionStream?